### ISSUE TITLE
Bad system call

### ISSUE THREAD

**Issue by PanagiotisSop** [NetworkConfiguration/dhcpcd#159]
This is a problem with version 9.4.1, when I built the latest source (commit 6797c08) the command seems to be working alright.

Distribution: Arch Linux
Arch: x86_64
Linux kernel version: 6.1.1
dhcpcd version: 9.4.1
dhcpcd compiled in features: INET ARP ARPing IPv4LL INET6 DHCPv6 AUTH PRIVSEP
glibc version: 2.36

When I issue the command `dhcpcd -U, --dumplease [interface]` a core dump with the following backtrace is generated: 
```
    Stack trace of thread 136124:
    #0  0x00007f63c62f45ae fstatat (libc.so.6 + 0xf65ae)
    #1  0x00007f63c62708f3 _IO_file_doallocate (libc.so.6 + 0x728f3)
    #2  0x00007f63c627e5d0 _IO_doallocbuf (libc.so.6 + 0x805d0)
    #3  0x00007f63c627d958 _IO_file_overflow (libc.so.6 + 0x7f958)
    #4  0x00007f63c627cafe _IO_file_xsputn (libc.so.6 + 0x7eafe)
    #5  0x00007f63c62595df n/a (libc.so.6 + 0x5b5df)
    #6  0x00007f63c631402b __printf_chk (libc.so.6 + 0x11602b)
    #7  0x0000557949ddd77a script_dump (dhcpcd + 0x1f77a)
    #8  0x0000557949dc9ae1 main (dhcpcd + 0xbae1)
    #9  0x00007f63c6221290 n/a (libc.so.6 + 0x23290)
    #10 0x00007f63c622134a __libc_start_main (libc.so.6 + 0x2334a)
    #11 0x0000557949dc9c5e _start (dhcpcd + 0xbc5e)
    ELF object binary architecture: AMD x86-64
```
I also get that in my system log in addition to the core dump:
```
    Dec 23 22:18:17 dhcpcd[151137]: control_free: No such file or directory
```
on the standard output, I get:
```
    Bad system call (core dumped)
```
or if I use the -d/--debug flag:
```
    send OK
    Bad system call (core dumped)
```
Here are the last 5 system calls as reported by `strace` before the crash (wlan0 is the name of my wireless interface):
```
    ppoll([{fd=3, events=POLLIN}], 1, {tv_sec=4, tv_nsec=999059199}, [], 8) = 1 ([{fd=3, revents=POLLIN}], left {tv_sec=4, tv_nsec=999048845})
    read(3, "\231\0\0\0\0\0\0\0", 8)        = 8
    ppoll([{fd=3, events=POLLIN}], 1, {tv_sec=4, tv_nsec=998335423}, [], 8) = 1 ([{fd=3, revents=POLLIN}], left {tv_sec=4, tv_nsec=998327041})
    read(3, "reason=CARRIER\0interface=wlan0\0p"..., 153) = 153
    newfstatat(1, "", {st_mode=025141332360, st_size=832, ...}, AT_EMPTY_PATH) = 262
    +++ killed by SIGSYS (core dumped) +++
```

Sometimes instead of crashing, I get the following message on the standard output:
```
    dhcpcd_readdumptimeout
```
or if I use the -d/--debug flag, I get this output:
```
    sandbox: seccomp
    send OK
    dhcpcd_readdumptimeout
```
and on the syslog I get:
```
    Dec 23 22:00:54 dhcpcd[115743]: ps_ctl_dispatch: cannot handle another client
    Dec 23 22:00:59 dhcpcd[115743]: control_free: No such file or directory
```

**Comment by rsmarples** [comment]
This should already be fixed in the dhcpcd-9 branch. Can you test it please? I aim to put a new release out this weekend.

**Comment by rsmarples** [comment]
Fixed in dhcpcd-9.5.0
